src/app/pages/users/user/stripe/payment-methods/payment-methods-table-data-source.ts

```diff
@@ -0,0 +1,231 @@
+          // Error
+          observer.error(error);
+        });
+      } else {
+        observer.next({
+          count: 0,
+          result: [],
+        });
+        observer.complete();
+      }
+    });
+  }
+
+  public buildTableDef(): TableDef {
+    return {
+      search: {
+        enabled: false,
+      },
+      hasDynamicRowAction: true,
+    };
+  }
+
+  public buildTableColumnDefs(): TableColumnDef[] {
+    const columns: TableColumnDef[] = [
+      {
+        id: 'status',
+        name: 'general.status',
+        isAngularComponent: true,
+        angularComponent: PaymentMethodStatusComponent,
+        headerClass: 'text-center col-15p',
+        class: 'text-center col-15p',
+      },
+      {
+        id: 'default',
+        name: 'general.default',
+        headerClass: 'text-center col-10p',
+        class: 'text-center col-10p',
+        formatter: (defaultPaymentMethod: boolean, paymentMethod: BillingPaymentMethod) => {
+          return paymentMethod.isDefault ?
+            this.translateService.instant('general.yes') : this.translateService.instant('general.no');
+        },
+      },
+    ]
+      if (!this.isAdmin) {
+        columns.push(
+          {
+            id: 'id',
+            name: 'general.id',
+            headerClass: 'text-center col-10p',
+            class: 'text-center col-10p',    
+          },
+        );
+      }
+      columns.push(
+        {
+          id: 'type',
+          name: 'settings.billing.payment_methods_type',
+          headerClass: 'text-center col-10p',
+          class: 'text-center col-10p capitalize',
+          },
+        {
+          id: 'brand',
+          name: 'settings.billing.payment_methods_brand',
+          headerClass: 'text-center col-15p',
+          class: 'text-center col-15p capitalize',
+          },
+        {
+          id: 'last4',
+          name: 'settings.billing.payment_methods_ending_with',
+          headerClass: 'text-center col-10p',
+          class: 'text-center col-10p',
+          },
+        {
+          id: 'expiringOn',
+          name: 'settings.billing.payment_methods_expiring_on',
+          headerClass: 'text-center col-10p',
+          class: 'text-center col-10p',
+          formatter: (expiringOn: Date) => {
+            return this.datePipe.transform(expiringOn, 'MM/YYYY');
+          },
+        },
+        {
+          id: 'createdOn',
+          name: 'general.created_on',
+          headerClass: 'text-center col-15p',
+          class: 'text-center col-15p',
+          formatter: (createdOn: Date) => this.datePipe.transform(createdOn)
+        },
+      )
+    return columns;
+  }
+
+  public buildTableActionsDef(): TableActionDef[] {
+    const tableActionsDef = super.buildTableActionsDef();
+    if (this.activatedRoute.snapshot.url[0]?.path === 'profile') {
+      this.currentUserID = this.centralServerService.getLoggedUser().id;
+    } else {
+      this.currentUserID = this.windowService.getSearch('userID');
```

Comment:
@LucasBrazi06  @ClaudeROSSI  is this the right way to handle the current user ?

Reply:
Nope. `this.windowService.getSearch()` is meant for building search filters. Ideally it should not be used for security reason and parameters transmission between components should be done by using @Input and @Output. The second line is a security issue: possible injection by manipulating window attributes client side.

`this.centralServerService.getLoggedUser().id` should be enough, I do not get why you need the second line.

Reply:
I need to know which user is currently shown in the user dialog (when Admin edit a user) to be able to handle the canCreatePaymentMethod()
Admin **is able** to list payment methods even if it's not its own
Admin **is not able** to create payment method if it's not its own
@ClaudeROSSI does the use case is correct ?

Reply:
So basically you need to pass a value from one component to another? 
Component one a user line in a tab -> component two the user billing details? 

If yes, you can do it without 'digging' the windows object key/values with Angular.

Reply:
I can try another solution : check the loggedUser is admin, if so he can only list if he's not in the 'profile' path
If in the profile path he can list **and** create payment method
(but don't know if checking we are in a specific path is a better solution ?)

I don't get the digging part in your comment, anyway for now I didn't get the solution to pass it differently and needed it to work but I will get a different approach or finding the 'not digging' approach ;)